### PR TITLE
[shim] Interpose `getsockopt()` and `setsockopt()`

### PR DESCRIPTION
**Pull request by ppenna**
## Description

- This PR closes https://github.com/microsoft/demikernel/issues/1272
- This PR closes https://github.com/microsoft/demikernel/issues/1273

### CODE REVIEW

File: shim/src/ctrl/getsockopt.c [hunk 1]
```diff
@@ -35,12 +36,35 @@ int __getsockopt(int sockfd, int level, int optname, void *optval, socklen_t *op
...
+    // Issue warnings for common options that are not supported.
+    if (level == SOL_SOCKET && optname == SO_KEEPALIVE)
+    {
+        WARN("%s is not supported", "SO_KEEPALIVE");
+    }
+    else if (level == SOL_SOCKET && optname == SO_REUSEADDR)
+    {
+        WARN("%s is not supported", "SO_REUSEADDR");
+    }
+    else if (level == IPPROTO_TCP && optname == TCP_NODELAY)
```
Comment (iyzhang) [inline shim/src/ctrl/getsockopt.c:48]: Is there a reason that TCP_NODELAY is not supported? We support it

Comment (ppenna) [reply shim/src/ctrl/getsockopt.c:48]: It is not supported by `demi_getsockopt()`:

https://github.com/microsoft/demikernel/blob/4b1ed79c2216d1e8c908292129193f9331fec517/src/rust/demikernel/bindings.rs#L863-L869

Comment (iyzhang) [reply shim/src/ctrl/getsockopt.c:48]: Oh, that is in my recent PR! Ha, you might want to wait to merge this until PR #1274 is merged.

**Review by iyzhang — APPROVED** [review]
Looks good to me. Please add support for SO_KEEPALIVE and SO_NODELAY after merging #1274

**Comment by ppenna** [comment]
> Looks good to me. Please add support for SO_KEEPALIVE and SO_NODELAY after merging #1274

I will handle this as a different PR, as #1274 is adding partial support for `SO_KEEPALIVE` and `SO_NODELAY`. For instance, #1274 should add:
- Fix documentation.
- Provide unit tests.

**Pull request merged**